src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +80,19 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        Predicate<Person> ifPersonHasSuitableData = person -> {
+            if (person.getSex() == Person.Sex.MAN
+                                && person.getAge() >= fromAge
+                                && person.getAge() <= maleToAge) {
+                return true;
+            }
+            return person.getSex() == Person.Sex.WOMAN
+                    && person.getAge() >= fromAge
+                    && person.getAge() <= femaleToAge;
+        };
+        return peopleList.stream()
+                .filter(ifPersonHasSuitableData)
+                .collect(Collectors.toList());
     }
```

Comment:
I had the same realization from the start, but then i found better varian. It is more short. Сonsider as an option)
```suggestion
    }return peopleList.stream()
                .filter(person -> {
                    if (person.getSex() == Person.Sex.MAN) {
                        return person.getAge() >= fromAge && person.getAge() <= maleToAge;
                    } else {
                        return person.getAge() >= fromAge && person.getAge() <= femaleToAge;
                    }
                })
                .collect(Collectors.toList());
```

Reply:
Frankly, I thought about the option as well 
Perhaps, let's not change it and ask a mentor about it, if you don't mind

Reply:
you can siplify your predicate and use there only one return statement